integrations/Braze/browser.js

```diff
@@ -124,46 +130,56 @@ class Braze {
+   *  address: {                                                                                      address: {
+   *   city: "Manali",                                                                                 city: "Shimla",
+   *   country: "India",                                                                               country: "India",
+   *  },                                                                                              },
+   * });                                                                                             });
+   * As both payload have same userId so it will deeply check all other attributes and pass the unique fields
+   * or the updated fields.
+   * @param {*} rudderElement
+   */
   identify(rudderElement) {
+    logger.debug("in Braze identify");
     const { userId } = rudderElement.message;
     const { address } = rudderElement.message.context.traits;
-    const { birthday } = rudderElement.message.context.traits;
+    const birthday =
+      rudderElement.message.context.traits?.birthday ||
+      rudderElement.message.context.traits?.dob;
     const { email } = rudderElement.message.context.traits;
-    const { firstname } = rudderElement.message.context.traits;
+    const firstname =
+      rudderElement.message.context.traits?.firstname ||
+      rudderElement.message.context.traits?.firstName;
     const { gender } = rudderElement.message.context.traits;
-    const { lastname } = rudderElement.message.context.traits;
+    const lastname =
+      rudderElement.message.context.traits?.lastname ||
+      rudderElement.message.context.traits?.lastName;
     const { phone } = rudderElement.message.context.traits;
 
-    // This is a hack to make a deep copy that is not recommended because it will often fail:
-    const traits = JSON.parse(
-      JSON.stringify(rudderElement.message.context.traits)
-    );
-
-    window.braze.changeUser(userId);
-    // method removed from v4 https://www.braze.com/docs/api/objects_filters/user_attributes_object#braze-user-profile-fields
-    // window.braze.getUser().setAvatarImageUrl(avatar);
-    if (email) window.braze.getUser().setEmail(email);
-    if (firstname) window.braze.getUser().setFirstName(firstname);
-    if (gender) window.braze.getUser().setGender(this.formatGender(gender));
-    if (lastname) window.braze.getUser().setLastName(lastname);
-    if (phone) window.braze.getUser().setPhoneNumber(phone);
-    if (address) {
-      window.braze.getUser().setCountry(address.country);
-      window.braze.getUser().setHomeCity(address.city);
-    }
-    if (birthday) {
-      window.braze
-        .getUser()
-        .setDateOfBirth(
-          birthday.getUTCFullYear(),
-          birthday.getUTCMonth() + 1,
-          birthday.getUTCDate()
-        );
-    }
-
     // remove reserved keys https://www.appboy.com/documentation/Platform_Wide/#reserved-keys
     const reserved = [
-      "avatar",
```

Comment:
Are these (avatar, first_name, last_name) no more reserved keys with current braze version ?

Reply:
We are not using these keywords previously here and also deleting these words from traits. So removed that from the list.